ca/config.go

```diff
@@ -472,18 +480,12 @@ func RenewTLSConfig(ctx context.Context, s *SecurityConfig, connBroker *connecti
 					return
 				}
 			} else {
-				// If we have an expired certificate, we let's stick with the starting default in
-				// the hope that this is a temporary clock skew.
+				// If we have an expired certificate, try to renew immediately: the hope that this is a temporary clock skew, or
```

Comment:
Oh wait, sorry for the ridiculous enumeration of the backoff, I think I misunderstood your comment:

>  it looks like we'd go into a renewal loop, since we reset the exponential backoff after each attempt when the certificate comes back expired.

If the certificate expires, then the renewal fails, since `RenewTLSConfigNow` calls `RequestAndSaveNewCertificates`, which [validates the cert, including expiry](https://github.com/docker/swarmkit/pull/1919/files#diff-3787a2543ed89ccb97fcdb3d8a17096aR175) before returning.  If it fails the expiry validation, `RequestAndSaveCertificates` and thus `RenewTLSConfigNow` fails, which means the exponential backoff doesn't get reset to 0.